File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/mr/knowledge/d/Knowledgedto.java
package iih.ci.mr.knowledge.d;

import xap.mw.core.data.*;
import xap.mw.coreitf.d.*;
import java.math.BigDecimal;

/**
 * 实体 DTO数据 
 * 
 */
public class Knowledgedto extends BaseDTO {
	private static final long serialVersionUID = 1L;
	
	/**
	 * 类别
	 * @return String
	 */
	public String getCategory() {
		return ((String) getAttrVal("Category"));
	}
	/**
	 * 类别
	 * @param Category
	 */
	public void setCategory(String Category) {
		setAttrVal("Category", Category);
	}
	/**
	 * 名称
	 * @return String
	 */
	public String getName() {
		return ((String) getAttrVal("Name"));
	}
	/**
	 * 名称
	 * @param Name
	 */
	public void setName(String Name) {
		setAttrVal("Name", Name);
	}
	/**
	 * 所属
	 * @return String
	 */
	public String getAttribute() {
		return ((String) getAttrVal("Attribute"));
	}
	/**
	 * 所属
	 * @param Attribute
	 */
	public void setAttribute(String Attribute) {
		setAttrVal("Attribute", Attribute);
	}
	/**
	 * 个人
	 * @return FBoolean
	 */
	public FBoolean getPersonal() {
		return ((FBoolean) getAttrVal("Personal"));
	}
	/**
	 * 个人
	 * @param Personal
	 */
	public void setPersonal(FBoolean Personal) {
		setAttrVal("Personal", Personal);
	}
	/**
	 * 科室
	 * @return FBoolean
	 */
	public FBoolean getAdministrative() {
		return ((FBoolean) getAttrVal("Administrative"));
	}
	/**
	 * 科室
	 * @param Administrative
	 */
	public void setAdministrative(FBoolean Administrative) {
		setAttrVal("Administrative", Administrative);
	}
	/**
	 * 所属用户
	 * @return String
	 */
	public String getId_user() {
		return ((String) getAttrVal("Id_user"));
	}
	/**
	 * 所属用户
	 * @param Id_user
	 */
	public void setId_user(String Id_user) {
		setAttrVal("Id_user", Id_user);
	}
	/**
	 * 知识内容
	 * @return String
	 */
	public String getKnowledge_content() {
		return ((String) getAttrVal("Knowledge_content"));
	}
	/**
	 * 知识内容
	 * @param Knowledge_content
	 */
	public void setKnowledge_content(String Knowledge_content) {
		setAttrVal("Knowledge_content", Knowledge_content);
	}
	/**
	 * 知识库编码
	 * @return String
	 */
	public String getId_knowledge() {
		return ((String) getAttrVal("Id_knowledge"));
	}
	/**
	 * 知识库编码
	 * @param Id_knowledge
	 */
	public void setId_knowledge(String Id_knowledge) {
		setAttrVal("Id_knowledge", Id_knowledge);
	}
	/**
	 * ds
	 * @return String
	 */
	public String getDs() {
		return ((String) getAttrVal("Ds"));
	}
	/**
	 * ds
	 * @param Ds
	 */
	public void setDs(String Ds) {
		setAttrVal("Ds", Ds);
	}
}

File: client/iih_stub/iih.ci_stub/iih.ci.mr/knowledge/d/Knowledgedto.cs
using System;
using xap.mw.core.data;
using xap.mw.coreitf.d;

namespace iih.ci.mr.knowledge.d
{
    /// <summary>
    /// Knowledgedto 的摘要说明。
    /// </summary>
    public class Knowledgedto : BaseDTO {

        public Knowledgedto() {
 
        }

        /// <summary>
        /// 类别
        /// </summary>
		public string Category {
            get { return getAttrVal<string>("Category",null); }
            set { setAttrVal<string>("Category", value); }
        }

        /// <summary>
        /// 名称
        /// </summary>
		public string Name {
            get { return getAttrVal<string>("Name",null); }
            set { setAttrVal<string>("Name", value); }
        }

        /// <summary>
        /// 所属
        /// </summary>
		public string Attribute {
            get { return getAttrVal<string>("Attribute",null); }
            set { setAttrVal<string>("Attribute", value); }
        }

        /// <summary>
        /// 个人
        /// </summary>
		public bool? Personal {
            get { return getAttrVal<FBoolean>("Personal",null); }
            set { setAttrVal<FBoolean>("Personal", value); }
        }

        /// <summary>
        /// 科室
        /// </summary>
		public bool? Administrative {
            get { return getAttrVal<FBoolean>("Administrative",null); }
            set { setAttrVal<FBoolean>("Administrative", value); }
        }

        /// <summary>
        /// 所属用户
        /// </summary>
		public string Id_user {
            get { return getAttrVal<string>("Id_user",null); }
            set { setAttrVal<string>("Id_user", value); }
        }

        /// <summary>
        /// 知识内容
        /// </summary>
		public string Knowledge_content {
            get { return getAttrVal<string>("Knowledge_content",null); }
            set { setAttrVal<string>("Knowledge_content", value); }
        }

        /// <summary>
        /// 知识库编码
        /// </summary>
		public string Id_knowledge {
            get { return getAttrVal<string>("Id_knowledge",null); }
            set { setAttrVal<string>("Id_knowledge", value); }
        }

        /// <summary>
        /// ds
        /// </summary>
		public string Ds {
            get { return getAttrVal<string>("Ds",null); }
            set { setAttrVal<string>("Ds", value); }
        }
        /// <summary>
        /// 返回属性列表
        /// </summary>
        /// <returns></returns>
        public override string[] getAttrNames()
        {
            return new string[]{ "Category", "Name", "Attribute", "Personal", "Administrative", "Id_user", "Knowledge_content", "Id_knowledge", "Ds"};
        }
        
        /// <summary>
        /// 返回全路径DO类名
        /// </summary>
        /// <returns></returns>
        public override string getFullClassName()
        {
            return "iih.ci.mr.knowledge.d.Knowledgedto";
        }
    }
}
